crates/txpool/src/lib.rs

```diff
@@ -1,10 +1,116 @@
-// Copyright 2021 Gnosis Ltd.
-// SPDX-License-Identifier: Apache-2.0
+// Copyright 2015-2018 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Generic Transaction Pool
+//!
+//! An extensible and performant implementation of Ethereum Transaction Pool.
+//! The pool stores ordered, verified transactions according to some pluggable
+//! `Scoring` implementation.
+//! The pool also allows you to construct a set of `pending` transactions according
+//! to some notion of `Readiness` (pluggable).
+//!
+//! The pool is generic over transactions and should make no assumptions about them.
+//! The only thing we can rely on is the `Scoring` that defines:
+//!  - the ordering of transactions from a single sender
+//!  - the priority of the transaction compared to other transactions from different senders
+//!
+//! NOTE: the transactions from a single sender are not ordered by priority,
```

Comment:
I suppose if the transactions are coming from a single sender, then the ordering should follow the `nonce` field in the `tx`. According to the standard, the transaction at `nonce + 1` is allowed to make assumptions about the outcome of the transaction marked with `nonce`.